### PR TITLE
[rearchitecture] Fix run_bot/butler

### PR DESCRIPTION
**Pull request by jonathanmetzman**
Print stacktraces properly.
Propogate errors so that butler command
fails when run_bot and other commands fails.
Related: #3008

### CODE REVIEW

**Review by hogo6002 — APPROVED** [review]

File: butler.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def _setup_args_for_remote(parser):
       '-i',
       '--instance-name',
       required=True,
-      help=('The instance name (e.g. clusterfuzz-linux-0005).'))
+      help='The instance name (e.g. clusterfuzz-linux-0005).')
```
Comment (Alan32Liu) [inline butler.py:58]: Very trivial nit: 'e.g.' -> 'e.g.,'.
Probably not worth fixing at all : )

File: src/local/butler/run_bot.py [hunk 2]
```diff
@@ -122,5 +123,8 @@ def _stop_handler(*_):
     signal.signal(signal.SIGTERM, _stop_handler)
     common.process_proc_output(proc)
     proc.wait()
+    exit_code = proc.wait()
```
Comment (Alan32Liu) [inline src/local/butler/run_bot.py:126]: Does this mean to overwrite the previous line, instead of appending to it?

Comment (jonathanmetzman) [reply src/local/butler/run_bot.py:126]: Nope. that was a mistake.

Comment (jonathanmetzman) [reply src/local/butler/run_bot.py:126]: https://github.com/google/clusterfuzz/pull/3037

**Pull request merged**